### PR TITLE
Code review and dockerization

### PR DESCRIPTION
**Pull request by patrickwestphal**
This merge request contains the following changes:

- Replace `print( )` statements with proper logging
- Removed redundant code/statements
- Use of data classes
- Better PEP-8 compliance
- Docker setup
- `FIXME` comments pointing to the issues
  - #18 
  - #19 
  - #20

### CODE REVIEW

**Comment by rudygarrido** [comment]
I think there was some bug fixing, documentation, modularization and the duplicate feature that reside in the development branch that could have some issues later on since the origin hadn't been merged yet. But I figure these can be addressed once we're fully back to working on the project 🤔 

Also, I'm not sure if I'm missing it somewhere but wouldn't the Dockerfile also need to introduce the files the logger is requiring?

File: main.py [hunk 1]
```diff
@@ -213,11 +354,12 @@ def zenodo(name, g, results):
...
-    serializable_results, results = find(name, results)
+    serializable_results = search_openalex.find(name, results)
     search_result = json.dumps(serializable_results)
+
     g.parse(data=search_result, format='json-ld')
-    return g, results
 
 
-demo.launch()
+demo.launch(server_name="0.0.0.0")
```
Comment (rudygarrido) [inline main.py:365]: Maybe there's a reason but why was this changed? It raising an error on my local PC but I guess it works fine on the Docker environment?

Comment (patrickwestphal) [reply main.py:365]: Oh, does it? I added it because without this, Gradio would only listen on 127.0.0.1, i.e. localhost. Then it would be of no use inside a Docker container as it cannot be accessed from the outside.

Comment (rudygarrido) [reply main.py:365]: I see! If it works on the Docker then all cool 😎
